pkg/services/serviceaccounts/manager/service.go

```diff
@@ -289,3 +292,38 @@ func validAPIKeyID(apiKeyID int64) error {
 	}
 	return nil
 }
+
+// hideApiKeysTabIfNoAPIKeysPresent is used to remove the apikeys tab from the admin tabs
+// if there are no apikeys, this makes it so that we only issue service account tokens instead
+func hideApiKeysTabIfNoAPIKeysPresent(l *log.ConcreteLogger, store *sqlstore.SQLStore, kvStore kvstore.KVStore) {
+	var err error
+	ctx := context.Background()
+	sess := store.GetSqlxSession()
+	orgs := make([]*org.OrgDTO, 0)
+	err = sess.Select(ctx, &orgs, "SELECT id, name FROM org")
+	if err != nil {
+		l.Error(fmt.Sprintf("unable to query for org for hiding api keys: %e", err))
+		return
+	}
+	if len(orgs) != 1 {
+		l.Info("do not want to hide apikeys for more than 1 org")
+		return
+	}
+
+	orgID := orgs[0].ID
+	apikeys := make([]*apikey.APIKey, 0)
+	if orgID != -1 {
+		err = sess.Select(ctx, &apikeys, "SELECT * FROM api_key where org_id =?", orgID)
+	}
+	if err != nil {
+		l.Error(fmt.Sprintf("could not query for apikeys in org: %e", err))
+		return
+	}
+	if len(apikeys) == 0 {
+		if err := kvStore.Set(ctx, orgID, "serviceaccounts", "hideApiKeys", "1"); err != nil {
```

Comment:
We have store method for this case
https://github.com/grafana/grafana/blob/c637a5543e9a40d35457744618cae003220be6c5/pkg/services/serviceaccounts/database/store.go#L407

Reply:
that's true. if a new org is created we need to hide the tab for that org. 🤔 let me see what we should do. Maybe we should move this into the org creation until we remove apikeys tab

Reply:
or the other way would be if this is set, we should then have a global flag that is checked on new org setup. Either way works.

Let me explore a bit